commerce-notification-test/src/testIntegration/java/com/liferay/commerce/notification/test/CommerceSubscriptionsNotificationTest.java

```diff
@@ -0,0 +1,277 @@
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Ignore;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * @author Luca Pellizzon
+ */
+@Ignore
+@RunWith(Arquillian.class)
+public class CommerceSubscriptionsNotificationTest {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new AggregateTestRule(
+			new LiferayIntegrationTestRule(),
+			PermissionCheckerTestRule.INSTANCE);
+
+	@Before
+	public void setUp() throws Exception {
+		_company = CompanyTestUtil.addCompany();
+
+		_user = UserTestUtil.addUser(_company);
+
+		_group = GroupTestUtil.addGroup(
+			_company.getCompanyId(), _user.getUserId(), 0);
+
+		_commerceCurrency = CommerceCurrencyTestUtil.addCommerceCurrency(
+			_group.getGroupId());
+
+		_serviceContext = ServiceContextTestUtil.getServiceContext(
+			_company.getCompanyId(), _group.getGroupId(), _user.getUserId());
+
+		_commerceChannelLocalService.addCommerceChannel(
+			_group.getGroupId(),
+			_group.getName(_serviceContext.getLanguageId()) + " Portal",
+			CommerceChannelConstants.CHANNEL_TYPE_SITE, null, StringPool.BLANK,
+			StringPool.BLANK, _serviceContext);
+
+		User user = UserTestUtil.addUser(
+			_user.getCompanyId(), _user.getUserId(), "businessUser",
+			_serviceContext.getLocale(), RandomTestUtil.randomString(),
+			RandomTestUtil.randomString(),
+			new long[] {_serviceContext.getScopeGroupId()}, _serviceContext);
+
+		_commerceAccount = CommerceAccountTestUtil.addBusinessCommerceAccount(
+			_user.getUserId(), RandomTestUtil.randomString(),
+			RandomTestUtil.randomString(), RandomTestUtil.randomString(),
+			new long[] {user.getUserId()}, null, _serviceContext);
+
+		CommerceAccountTestUtil.addCommerceAccountGroupAndAccountRel(
+			_company.getCompanyId(), RandomTestUtil.randomString(),
+			CommerceAccountConstants.ACCOUNT_GROUP_TYPE_STATIC,
+			_commerceAccount.getCommerceAccountId(), _serviceContext);
+
+		_addCommerceNotificationTemplates();
+	}
+
+	@After
+	public void tearDown() throws PortalException {
+		for (CommerceNotificationTemplate commerceNotificationTemplate :
+				_commerceNotificationTemplateList) {
+
+			_commerceNotificationTemplateLocalService.
+				deleteCommerceNotificationTemplate(
+					commerceNotificationTemplate);
+		}
+	}
+
+	@Test
+	public void testSubscriptionStatusNotification() throws Exception {
+		_commerceOrder = CommerceTestUtil.addB2CCommerceOrder(
+			_group.getGroupId(), _user.getUserId(),
+			_commerceCurrency.getCommerceCurrencyId());
+
+		_commerceOrder = CommerceTestUtil.addCheckoutDetailsToUserOrder(
+			_commerceOrder, _user.getUserId(), true);
+
+		_commerceOrder = CommerceTestUtil.checkoutOrder(_commerceOrder);
+
+		_commerceOrderLocalService.setCommerceOrderToTransmit(
+			_commerceOrder.getUserId(), _commerceOrder);
+
+		Thread.sleep(5000);
```

Comment:
Should this be 5 seconds, or 1 second like elsewhere?

Reply:
It could, but for now this test is ignored, because CommerceSubscriptionStatusMessageListener is not working properly. When we will understand why, we'll be able to fine-tune the test